### ISSUE TITLE
Move script part of .travis.yml into a script file

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap-docker-pkg-tests#1]
The `script` part of the .travis.yml contains a super long last line, containign this code (as of right now):

```
docker run -t gapsystem/gap-docker /bin/sh -c "cd /home/gap/inst/gap4r7/; sudo wget https://raw.githubusercontent.com/gap-system/pkg-tests/master/testpackage.g; sudo wget https://raw.githubusercontent.com/gap-system/pkg-tests/master/testpackage.sh; sudo chmod u+x testpackage.sh ; sudo ./testpackage.sh ${PKG_NAME}; cat dev/log/*; if [ \`cat dev/log/*.${PKG_NAME} | grep -c '#I  No errors detected while testing package'\` != '3' ]; then exit 1; fi;"
```

I would recommend moving the code after `-c` into a separate shell script, and then change .travis.yml to execute that shell script. That script then might look like this:

``` sh
cd /home/gap/inst/gap4r7/
sudo wget https://raw.githubusercontent.com/gap-system/pkg-tests/master/testpackage.g
sudo wget https://raw.githubusercontent.com/gap-system/pkg-tests/master/testpackage.sh
sudo chmod u+x testpackage.sh
sudo ./testpackage.sh ${PKG_NAME}
cat dev/log/*
if [ \`cat dev/log/*.${PKG_NAME} | grep -c '#I  No errors detected while testing package'\` != '3' ]; then
 exit 1;
fi;
```

This has various advantages:
- it becomes much easier for a human being to understand what is going on (for example, I can now formulate questions like "why does this download scripts from `gap-system/pkg-tests`, instead of using those same script included in this repository?")
- it allows you to add comments to the code, if needed
- changes to the script code will be easier to see
- if you start the script with `#!/bin/sh -ev` or `#!/bin/sh -ex` then the docker logs will become more useful, too

**Comment by alex-konovalov** [comment]
Thanks, agree that this is a good idea.

>"why does this download scripts from gap-system/pkg-tests, instead of using those same script included in this repository?")

Because it was formerly named `pkg-tests` and then renamed to `gap-docker-pkg-tests` (there may be other `pkg-tests` later). I will update URLs.

**Comment by fingolfin** [comment]
You should not hardcode those URLs at all, otherwise people cannot easily clone this repository (as the clone would still use the URLs pointing to the original repository).

Indeed, you do not need to wget those files at all: They are alread present, as the very first thing travis does is to clone this repository. Just look at the travis logs:
```
git clone --depth=50 --branch=master https://github.com/gap-system/pkg-tests.git
```

**Comment by fingolfin** [comment]
Ahah wait, the above is of course nonsense: The file is available to travis, but not in the docker image.

Hmm.. but perhaps one can copy it into the docker image somehow? Or, failing that, perhaps one can at least insert the correct URL into the script (instead of hard coding it)... I'll think about it.

**Comment by fingolfin** [comment]
According to docker docs, the `docker cp` command should be able to solve this. It can be used like this:
```
docker cp foo.txt mycontainer:/foo.txt
```
So I am guessing we would need:
```
docker cp testpackage.g gapsystem/gap-docker:/home/gap/inst/gap4r7/
docker cp testpackage.sh gapsystem/gap-docker:/home/gap/inst/gap4r7/
```

**Comment by alex-konovalov** [comment]
Hmm... Isn't `sudo wget https://raw.githubusercontent.com/gap-system/pkg-tests/master/testpackage.sh` doing exactly "perhaps one can copy it into the docker image somehow?" I think I need only to update the URL and have a shell script for all task as per original advice....

**Comment by fingolfin** [comment]
The point is: If I clone `gap-system/pkg-tests` to `fingolfin/pkg-tests`, it will still use the `gap-system/pkg-tests` version of the testpackage.* files. I would have to modify the .travis.yml file to fix that (and might easily forgot to do so, potentially leading to strange results).
And if you rename the repository (for example, from `pkg-tests` to `gap-docker-pkg-tests` -- oh wait, you already did that!), then the URLs become broken.

Thirdly, if one ever wants to have pull requests on this very repository, then these would still use the testpackage.* files from the master branch, instead of the one included in the pull request, which gives potentially misleading results (as the wrong thing is tested).

**Comment by fingolfin** [comment]
Ping?

**Comment by alex-konovalov** [comment]
@fingolfin : ping accepted :)

**Comment by alex-konovalov** [comment]
I am mostly waiting for GAP 4.8 public release and updated Docker container providing it - there, all ReadTest calls will be delegated to Test so I will have visible `########> Diff` markers for all diffs. Also, more packages there will comply with `#I  Errors detected while testing` and #I  No errors detected while testing` messages which I will be looking for.

**Comment by alex-konovalov** [comment]
Remind to myself: https://travis-ci.org/gap-system/gap-docker-master-testsuite/ is an example of organising this better.

**Comment by alex-konovalov** [comment]
In PR #2 I have simplified the way the test is invoked: 
- mounted the directory with the clone of the repository into the container
- created ci.sh script
- added HELP.md to display explanation for packages with failed tests.

This addresses #1, and, when finalised, may be propagated to other (currently 4) travis builds for packages passing/failing tests in the master and stable branches.

**Comment by alex-konovalov** [comment]
Closed by #2.
